pkgs/development/interpreters/python/hooks/requirements-check-hook.sh

```diff
@@ -0,0 +1,15 @@
+# Setup hook for requirements.
+echo "Sourcing requirements-check-hook"
+
+requirementsCheckPhase() {
+    if [ -z "$requirementsCheckFile" ]; then
+        echo "Error: no $requirementsCheckFile was given."
+        exit 1
+
+    @pythonCheckInterpreter@ @requirementsCheckScript@ $requirementsCheckFile
```

Comment:
Hardcoding the interpreter I don't like very much because it makes it harder when testing a Python env. In that case, one would have to set `PYTHONPATH` or `NIX_PYTHONPATH` to include that env.

The only alternative I see is using that env's Python to execute this, but we can't automate that, which means users would have to copy in the `check-requirements.py` script instead.